core/server/services/auth/api-key/admin.js

```diff
@@ -0,0 +1,112 @@
+ * 3. find a matching api_key record
+ * 4. verify the JWT (matching secret, matching URL path, not expired)
+ * 5. place the api_key object on `req.api_key`
+ *
+ * There are some specifcs of the JWT that we expect:
+ * - the "Key ID" header parameter should be set to the id of the api_key used to sign the token
+ *   https://tools.ietf.org/html/rfc7515#section-4.1.4
+ * - the "Audience" claim should match the requested API path
+ *   https://tools.ietf.org/html/rfc7519#section-4.1.3
+ */
+const authenticateAdminApiKey = function authenticateAdminApiKey(req, res, next) {
+    // we don't have an Authorization header so allow fallthrough to other
+    // auth middleware or final "ensure authenticated" check
+    if (!req.headers || !req.headers.authorization) {
+        return next();
+    }
+
+    const token = _extractTokenFromHeader(req.headers.authorization);
+
+    if (!token) {
+        return next(new common.errors.UnauthorizedError({
+            message: common.i18n.t('errors.middleware.auth.incorrectAuthHeaderFormat'),
+            code: 'INVALID_AUTH_HEADER'
+        }));
+    }
+
+    const decoded = jwt.decode(token, {complete: true});
+
+    if (!decoded || !decoded.header) {
+        return next(new common.errors.BadRequestError({
+            message: common.i18n.t('errors.middleware.auth.invalidToken'),
+            code: 'INVALID_JWT'
+        }));
+    }
+
+    const apiKeyId = decoded.header.kid;
+
+    models.ApiKey.findOne({id: apiKeyId}).then((apiKey) => {
+        if (!apiKey) {
+            return next(new common.errors.UnauthorizedError({
+                message: common.i18n.t('errors.middleware.auth.unknownAdminApiKey'),
+                code: 'UNKNOWN_ADMIN_API_KEY'
+            }));
+        }
+
+        if (apiKey.get('type') !== 'admin') {
+            return next(new common.errors.UnauthorizedError({
+                message: common.i18n.t('errors.middleware.auth.invalidApiKeyType'),
+                code: 'INVALID_API_KEY_TYPE'
+            }));
+        }
+
+        const secret = Buffer.from(apiKey.get('secret'), 'hex');
+        // ensure the token was meant for this endpoint
+        const options = Object.assign({
+            aud: req.originalUrl
+        }, JWT_OPTIONS);
+
+        try {
+            jwt.verify(token, secret, options);
+        } catch (err) {
+            if (err.name === 'TokenExpiredError' || err.name === 'JsonWebTokenError') {
+                return next(new common.errors.UnauthorizedError({
+                    message: common.i18n.t('errors.middleware.auth.invalidTokenWithMessage', {message: err.message}),
+                    code: 'INVALID_JWT',
+                    err
+                }));
+            }
+
+            // unknown error
+            return next(new common.errors.InternalServerError({err}));
+        }
+
+        // authenticated OK, store the api key on the request for later checks and logging
+        req.api_key = apiKey;
```

Comment:
Attention:

We attach `req.api_key` to `frame.options.context.api_key_id` (see e.g. shared/http)
And some input serializers use this attribute currently to check if an api key exists and if so, they assume it's a content API access. This no longer works with api key being an admin key?

Reply:
No that will no longer work. Seeing as having serializers differentiating between content and admin API is something we're doing in a lot of places. Does it make sense to have some simple middleware such as
```javascipt
router.use((req, res, next) => {
  req.contentApi = true;
  next();
});
```
in the content api routes? and have everything switch on that? that way it is explicit? maybe that doesn't play too well with the rendering. hmmm :thinking: 

How about we *always* assume Content API - this is safe.

Then if we have `req.isAdminAPI` Then we know it's admin api - that's super specific?

Reply:
If we would solve [this](https://github.com/TryGhost/Ghost/issues/10099#issuecomment-436160395), then the permission context parser can detect if the api key is type content or admin?

type content = public access
type admin = ?